### ISSUE TITLE
Must link bank twice to answer question or device pin

### ISSUE THREAD

**Issue by javierjulio** [synapsepayments/synapse_pay-ruby#6]
@joncalhoun @sankaet from testing the bank link API in the console (sandbox mode) I'm realizing that if I want to answer a question or provide device code I need to the username and password to login a second time just to get the MFA instance so I can call the `answer` method.

Is there no way to get the MFA instance without having to login a second time? I ask because developing a mobile app and an API back end, its async. I can't keep the MFA instance in memory as our API is stateless. So I login once to get the question or send code but then would need to login again to provide the answer. We want to avoid having to store in memory the users' bank account username and password in our mobile app but I don't see any other way around that.

To sum it up, once you link a bank account, is there no separate API to retrieve the MFA instance again so I can provide the answer in async form? I won't do that all in one request.

**Comment by joncalhoun** [comment]
You would likely want to store the user's `oauth_consumer_key` and `refresh_token` in your db, and then use those to create a client like so:

``` ruby
client = SynapsePay::Client.new("oauth_consumer_key", "refresh_token")
```

It sounds like we need to create an MFA method that doesn't require an MFA instance, which I will add to our todo list, but in the meantime you can create an MFA instance like so:

``` ruby
# Get the MFA on the first request
mfa = client.banks.link({
  :username => "synapse_code",
  :password => "test1234",
  :bank => "Bank of America"
})

# save the access token for the user and the bank (or pass it to the app and have it as part of the mfa answer response)
mfa.access_token 
```

Now on your next call (after you lose the mfa instance), you will need to reconstruct the client and the mfa instance like so:

``` ruby
client = SynapsePay::Client.new("oauth_consumer_key", "refresh_token")
mfa = SynapsePay:: BankMfaQuestions.new({ access_token: "access-token" }, nil, client)
# or mfa = SynapsePay:: BankMfaDevice.new({ access_token: "access-token" }, nil, client)
banks = mfa.answer("bank name", "answer")
```

The instance (mfa in this case) _must_ have the client as the third argument when creating it, because this client is used to make API calls using your user's `oauth_consumer_key`. If that ever expires, you use the `refresh_token` to get a new one. 

In your use case, where you want to answer an MFA in a separate web request, what we will likely add is something like:

``` ruby
banks = client.mfa.answer("access-token", "bank-name", "answer")
```

Which would allow you to avoid creating another MFA instance, but you would still need to store the user's `oauth_consumer_key` and `refresh_token` to access their account (or user `force_create = "no"` in the user create endpoint)
